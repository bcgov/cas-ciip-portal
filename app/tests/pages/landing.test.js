import React from 'react';
import Landing from '../../pages/landing';
import { mount } from 'enzyme';
import { render, fireEvent, cleanup } from '@testing-library/react';



describe('landing', () => {
    afterEach(cleanup);
    it('It matches the last accepted Snapshot', () => {
        const wrapper = mount(<Landing />);
        expect(wrapper).toMatchSnapshot();
    });

    it('get BCEID button should bring up modal', () => {
        const { getByTestId } = render(<Landing />);
        fireEvent.click(getByTestId('get-modal'));
        expect(getByTestId('modal')).toBeDefined();
        expect(getByTestId('btn-get-portal')).toBeDefined();
    });
    it('continue button should collapse modal and bring up bceid site', () => {
        const { getByTestId } = render(<Landing />);
        fireEvent.click(getByTestId('get-modal'));
        fireEvent.click(getByTestId('btn-get-portal'));
        expect({ getByTestId }).not.toContain('modal');
        expect(document.querySelector('.btn-portal').getAttribute("href")).toBe("https://www.bceid.ca/");
    });
});