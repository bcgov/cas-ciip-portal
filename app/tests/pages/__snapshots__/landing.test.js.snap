// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`landing It matches the last accepted Snapshot 1`] = `
<div>
  Array [
    "",
    <header
      className="jsx-1956122375"
    >
      <div
        className="jsx-1956122375 banner"
      >
        <a
          alt="British Columbia"
          className="jsx-1956122375"
          href="/"
        >
          <img
            alt="Go to the Government of British Columbia website"
            className="jsx-1956122375"
            src="/static/logo-banner.png"
          />
        </a>
        <h1
          className="jsx-1956122375"
        >
          CleanBC Industrial Incentive Program
        </h1>
      </div>
      <div
        className="buttons"
      >
        <ForwardRef
          noGutters={false}
        >
          <div
            className="row"
          >
            <Col
              md={
                Object {
                  "span": 8,
                }
              }
            >
              <div
                className="col-md-8"
              >
                <a
                  className="link"
                  href="/userProfile"
                >
                  <p
                    style={
                      Object {
                        "fontWeight": "bolder",
                        "marginTop": "13px",
                      }
                    }
                  >
                    Mary Jane
                     
                    <br />
                    Admin Rio Tinto
                  </p>
                </a>
              </div>
            </Col>
            <Col
              md={
                Object {
                  "span": 4,
                }
              }
            >
              <div
                className="col-md-4"
              >
                <ButtonToolbar
                  role="toolbar"
                >
                  <div
                    className="btn-toolbar"
                    role="toolbar"
                  >
                    <a
                      className="btn"
                      href="/applications"
                      style={
                        Object {
                          "backgroundColor": "#EDA500",
                          "color": "white",
                          "marginTop": "18px",
                        }
                      }
                    >
                      Logout
                    </a>
                  </div>
                </ButtonToolbar>
              </div>
            </Col>
          </div>
        </ForwardRef>
      </div>
    </header>,
  ]
  <div
    style={
      Object {
        "backgroundColor": "#f5f5f5",
        "borderBottomStyle": "solid",
        "borderWidth": "1px",
        "opacity": 1,
        "paddingBottom": "60px",
        "paddingTop": "60px",
        "width": "100%",
      }
    }
  >
    <h1
      style={
        Object {
          "color": "#003366",
          "fontSize": "Bold 25px",
          "fontWeight": "bold",
          "marginLeft": "400px",
        }
      }
    >
      Welcome to your CIIP Portal
    </h1>
    <p
      style={
        Object {
          "color": "black",
          "fontSize": "14px",
          "marginLeft": "400px",
          "width": "50%",
        }
      }
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
    </p>
  </div>
  <Container
    fluid={false}
  >
    <div
      className="container"
    >
      <ForwardRef
        noGutters={false}
      >
        <div
          className="row"
        >
          <Col
            md={true}
          >
            <div
              className="col-md"
            >
              <Card
                body={false}
                style={
                  Object {
                    "borderRadius": "15px",
                    "marginBottom": "25px",
                    "marginTop": "50px",
                    "textAlign": "center",
                    "width": "450px",
                  }
                }
              >
                <div
                  className="card"
                  style={
                    Object {
                      "borderRadius": "15px",
                      "marginBottom": "25px",
                      "marginTop": "50px",
                      "textAlign": "center",
                      "width": "450px",
                    }
                  }
                >
                  <CardBody
                    style={
                      Object {
                        "backgroundColor": "#003366",
                        "borderRadius": "15px",
                        "padding": "50px",
                      }
                    }
                  >
                    <div
                      className="card-body"
                      style={
                        Object {
                          "backgroundColor": "#003366",
                          "borderRadius": "15px",
                          "padding": "50px",
                        }
                      }
                    >
                      <CardTitle
                        style={
                          Object {
                            "color": "white",
                            "fontSize": "25px",
                            "fontWeight": "bold",
                            "marginBottom": "15px",
                          }
                        }
                      >
                        <ForwardRef
                          className="card-title"
                          style={
                            Object {
                              "color": "white",
                              "fontSize": "25px",
                              "fontWeight": "bold",
                              "marginBottom": "15px",
                            }
                          }
                        >
                          <div
                            className="card-title h5"
                            style={
                              Object {
                                "color": "white",
                                "fontSize": "25px",
                                "fontWeight": "bold",
                                "marginBottom": "15px",
                              }
                            }
                          >
                            Login with BCEID
                          </div>
                        </ForwardRef>
                      </CardTitle>
                      <CardText
                        style={
                          Object {
                            "color": "white",
                          }
                        }
                      >
                        <p
                          className="card-text"
                          style={
                            Object {
                              "color": "white",
                            }
                          }
                        >
                          Applying for CIIP requires having a valid BCEID. If you have already acquired yours click below to Login
                        </p>
                      </CardText>
                      <Button
                        active={false}
                        disabled={false}
                        size="lg"
                        style={
                          Object {
                            "backgroundColor": "#EDA500",
                            "color": "#003366",
                          }
                        }
                        type="button"
                        variant="primary"
                      >
                        <button
                          className="btn btn-primary btn-lg"
                          disabled={false}
                          style={
                            Object {
                              "backgroundColor": "#EDA500",
                              "color": "#003366",
                            }
                          }
                          type="button"
                        >
                          Login with BCEID
                        </button>
                      </Button>
                    </div>
                  </CardBody>
                </div>
              </Card>
            </div>
          </Col>
          <Col>
            <div
              className="col"
            >
              <Card
                body={false}
                style={
                  Object {
                    "borderRadius": "15px",
                    "marginBottom": "25px",
                    "marginTop": "50px",
                    "textAlign": "center",
                    "width": "450px",
                  }
                }
              >
                <div
                  className="card"
                  style={
                    Object {
                      "borderRadius": "15px",
                      "marginBottom": "25px",
                      "marginTop": "50px",
                      "textAlign": "center",
                      "width": "450px",
                    }
                  }
                >
                  <CardBody
                    style={
                      Object {
                        "padding": "50px",
                      }
                    }
                  >
                    <div
                      className="card-body"
                      style={
                        Object {
                          "padding": "50px",
                        }
                      }
                    >
                      <CardTitle
                        style={
                          Object {
                            "color": "#003366",
                            "fontSize": "25px",
                            "fontWeight": "bold",
                            "marginBottom": "15px",
                          }
                        }
                      >
                        <ForwardRef
                          className="card-title"
                          style={
                            Object {
                              "color": "#003366",
                              "fontSize": "25px",
                              "fontWeight": "bold",
                              "marginBottom": "15px",
                            }
                          }
                        >
                          <div
                            className="card-title h5"
                            style={
                              Object {
                                "color": "#003366",
                                "fontSize": "25px",
                                "fontWeight": "bold",
                                "marginBottom": "15px",
                              }
                            }
                          >
                            Get a BCEID
                          </div>
                        </ForwardRef>
                      </CardTitle>
                      <CardText
                        style={
                          Object {
                            "color": "#003366",
                          }
                        }
                      >
                        <p
                          className="card-text"
                          style={
                            Object {
                              "color": "#003366",
                            }
                          }
                        >
                          Applying for CIIP requires having a valid BCEID. Click here to acquire a BCEID. Once you have it, you can come to this page and proceed.
                        </p>
                      </CardText>
                      <Button
                        active={false}
                        data-testid="get-modal"
                        disabled={false}
                        onClick={[Function]}
                        size="lg"
                        style={
                          Object {
                            "backgroundColor": "#003366",
                          }
                        }
                        type="button"
                        variant="primary"
                      >
                        <button
                          className="btn btn-primary btn-lg"
                          data-testid="get-modal"
                          disabled={false}
                          onClick={[Function]}
                          style={
                            Object {
                              "backgroundColor": "#003366",
                            }
                          }
                          type="button"
                        >
                          Get a BCEID
                        </button>
                      </Button>
                    </div>
                  </CardBody>
                </div>
              </Card>
            </div>
          </Col>
        </div>
      </ForwardRef>
      <ForwardRef
        noGutters={false}
      >
        <div
          className="row"
        >
          <Col>
            <div
              className="col"
            >
              <h2
                style={
                  Object {
                    "color": "#003366",
                    "fontSize": "25px",
                    "fontWeight": "bold",
                    "marginBottom": "46px",
                    "marginTop": "15px",
                  }
                }
              >
                Apply for CIIP 2019
              </h2>
              <p
                style={
                  Object {
                    "marginBottom": "50px",
                    "width": "468px",
                  }
                }
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
              </p>
            </div>
          </Col>
          <Col>
            <div
              className="col"
            >
              <h2
                style={
                  Object {
                    "color": "#003366",
                    "fontSize": "25px",
                    "fontWeight": "bold",
                    "marginBottom": "46px",
                    "marginTop": "15px",
                  }
                }
              >
                Why do I need a BCEID
              </h2>
              <p
                style={
                  Object {
                    "marginBottom": "50px",
                    "width": "468px",
                  }
                }
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
              </p>
            </div>
          </Col>
        </div>
      </ForwardRef>
    </div>
  </Container>
  <footer
    className="jsx-55582602 footer"
  >
    <div
      className="jsx-55582602 container"
    >
      <ul
        className="jsx-55582602"
      >
        <li
          className="jsx-55582602"
        >
          <a
            className="jsx-55582602"
            href="."
          >
            Home
          </a>
        </li>
        <li
          className="jsx-55582602"
        >
          <a
            className="jsx-55582602"
            href="."
          >
            Disclaimer
          </a>
        </li>
        <li
          className="jsx-55582602"
        >
          <a
            className="jsx-55582602"
            href="."
          >
            Privacy
          </a>
        </li>
        <li
          className="jsx-55582602"
        >
          <a
            className="jsx-55582602"
            href="."
          >
            Accessibility
          </a>
        </li>
        <li
          className="jsx-55582602"
        >
          <a
            className="jsx-55582602"
            href="."
          >
            Copyright
          </a>
        </li>
        <li
          className="jsx-55582602"
        >
          <a
            className="jsx-55582602"
            href="."
          >
            Contact Us
          </a>
        </li>
      </ul>
    </div>
  </footer>
</div>
`;
